### PR TITLE
Allow building a subset of services.

### PR DESCRIPTION
**Pull request by chouseknecht**
##### ISSUE TYPE
Feature Pull Request

##### SUMMARY

Adds `service` option to the `build` command. This makes it possible to circumvent the orchestrated build and only build a subset of services.

### CODE REVIEW

File: container/engine.py [hunk 1]
```diff
@@ -302,6 +303,10 @@ def cmdrun_build(base_path, engine_name, flatten=True, purge_last=True, local_bu
         logger.info('Starting %s engine to build your images...'
                     % engine_obj.orchestrator_name)
         touched_hosts = engine_obj.hosts_touched_by_playbook()
+        if service:
+            touched_hosts = list(set(touched_hosts).intersection(service))
```
Comment (pilou-) [inline container/engine.py]:
```
touched_hosts = set(engine_obj.hosts_touched_by_playbook())
if service:
    touched_hosts &= service
    if not touched_hosts:
        raise AnsibleContainerHostNotTouchedByPlaybook()
```
could be used instead.

File: container/docker/engine.py [hunk 2]
```diff
@@ -429,7 +430,15 @@ def _fix_volumes(self, service_name, service_config):
     def get_config_for_build(self):
         compose_config = config_to_compose(self.config)
         orchestrated_hosts = self.hosts_touched_by_playbook()
+        if self.params.get('service'):
+            # only build a subset of the orchestrated hosts
+            orchestrated_hosts = list(set(orchestrated_hosts).intersection(self.params['service']))
+            for host in list(set(compose_config.keys()) - set(orchestrated_hosts)):
```
Comment (chouseknecht) [inline container/docker/engine.py]: Fixed.

File: container/docker/engine.py [hunk 3]
```diff
@@ -429,7 +430,15 @@ def _fix_volumes(self, service_name, service_config):
...
+                del compose_config[host]
+            if len(compose_config.keys()) == 0:
```
Comment (chouseknecht) [inline container/docker/engine.py]: Fixed.

File: container/engine.py (hunk 1 above)
Comment (chouseknecht) [inline container/engine.py]: Fixed

**Pull request merged**